# Ejercicio 24
Calcule el porcentaje de nulos para todas las columnas de geo_tags.csv 

In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')
#df = pd.read_csv('/home/andrezszambrano/Documents/Facultad de Ingeniería de Buenos Aires/Organización de Datos/Trabajos Prácticos /TP1/WikipediaDataset/contents.csv', nrows = 100)
df = pd.read_csv('./WikipediaDataset/geo_tags.csv')

In [2]:
df.head()

gt_id  gt_page_id gt_globe  gt_primary   gt_lat   gt_lon  gt_dim gt_type  \
0   158041     4328020    earth           0  41.5075  -5.8175    1000     NaN   
1  5844377     4670424    earth           0  13.7380 -89.2920   10000    city   
2  5844381     4670424    earth           0  13.7020 -89.2080   10000    city   
3  5844383     4670424    earth           0  13.4000 -89.7500   10000    city   
4  5844388     4670424    earth           0  12.8500 -88.0000   10000    city   

  gt_name gt_country gt_region  
0     NaN        NaN       NaN  
1     NaN        NaN       NaN  
2     NaN        NaN       NaN  
3     NaN        NaN       NaN  
4     NaN        NaN       NaN

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 556393 entries, 0 to 556392
Data columns (total 11 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   gt_id       556393 non-null  int64  
 1   gt_page_id  556393 non-null  int64  
 2   gt_globe    556393 non-null  object 
 3   gt_primary  556393 non-null  int64  
 4   gt_lat      556393 non-null  float64
 5   gt_lon      556393 non-null  float64
 6   gt_dim      556393 non-null  int64  
 7   gt_type     449893 non-null  object 
 8   gt_name     72557 non-null   object 
 9   gt_country  79192 non-null   object 
 10  gt_region   3088 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 46.7+ MB


In [4]:
df.shape

(556393, 11)

Hay 556393 filas en total. Procedo a calcular columna por columna la cantidad de nulos que contienen.

In [5]:
def calcularPorcentajeNulosDeColumna(cantTotalDeNoNulos):
    return (1 - (cantTotalDeNoNulos / 556393))

#### Columna gt_id

In [6]:
cantFilasConInfo = df['gt_id'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_page_id

In [7]:
cantFilasConInfo = df['gt_page_id'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_globe

In [9]:
cantFilasConInfo = df['gt_globe'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_primary

In [10]:
cantFilasConInfo = df['gt_primary'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_lat

In [12]:
cantFilasConInfo = df['gt_lat'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_lon

In [13]:
cantFilasConInfo = df['gt_lon'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_dim

In [14]:
cantFilasConInfo = df['gt_dim'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.0


#### Columna gt_type

In [16]:
cantFilasConInfo = df['gt_type'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.19141146635561557


#### Columna gt_name 

In [17]:
cantFilasConInfo = df['gt_name'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.8695939740435268


#### Columna gt_country

In [18]:
cantFilasConInfo = df['gt_country'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.8576689498250337


#### Columna gt_region

In [19]:
cantFilasConInfo = df['gt_region'].count()
print(calcularPorcentajeNulosDeColumna(cantFilasConInfo))

0.9944499661210691


## Conclusiones
Se puede concluir que hay atributos que un geo_tag debe tener por definición, por lo que la cantidad de nulos en esa columna es cero, quedando entonces un porcentaje de nulos igual a cero. Hay atributos más prescindibles, por lo que pueden hay geo_tags que no tienen dicho atributo, comenzando por el gt_type, con porcentaje 0.1914 de nulos, siguiendo por gt_name, gt_country, gt_region, que tienen un porcentaje de nulos de 0.8696, 0.8577, 0.8577, respectivamente.